source/beamline/IDEAS/IDEASBeamline.cpp

```diff
@@ -296,6 +296,16 @@ void IDEASBeamline::onShutterValueChanged()
 	emit overallShutterStatus(safetyShutter_->isOpen() && photonShutter2_->isOpen() && safetyShutter2_->isOpen());
 }
 
+AMXRFDetector *IDEASBeamline::xrfDetector() const
+{
+	if (ketek_ )
```

Comment:
As mentioned above, this function was designed return a default XRF detector. The specific XRF detector is return in another function:
`AMXRFDetector *IDEASBeamline::xrfDetector(IDEAS::FluorescenceDetectors detectorType)`